normandy/control/static/control/js/components/RecipeHistory.jsx

```diff
@@ -30,32 +30,64 @@ class RecipeHistory extends React.Component {
     const { revisionLog } = this.state;
     const { recipeId, recipe, dispatch } = this.props;
     return (
-      <div className="fluid-8">
+      <div className="fluid-8 recipe-history">
         <h3>Viewing revision log for: <b>{recipe ? recipe.name : ''}</b></h3>
         <ul>
-            {
-              this.state.revisionLog.map((revision, index) =>
-                <li key={revision.date_created} onClick={(e) => {
-                  let revisionInfo = {};
-                  if (index !== 0) {
-                    revisionInfo = {
-                      query: { revisionId: `${revision.id}` },
-                      state: { selectedRevision: revision.recipe }
-                    }
-                  }
-                  dispatch(push({
-                    pathname: `/control/recipe/${recipeId}/`,
-                    ...revisionInfo
-                  }))
-                }}><p className="revision-number">#{revision.recipe.revision_id} </p>
-                  <p><span className="label">Created On:</span>{ moment(revision.date_created).format('MMM Do YYYY - h:mmA') }</p>
-                </li>
-              )
-            }
+            {this.state.revisionLog.map((revision, index) =>
+              <HistoryItem
+                key={revision.id}
+                revision={revision}
+                recipe={recipe}
+                dispatch={dispatch}
+              />
+            )}
         </ul>
       </div>
     )
   }
 }
 
+class HistoryItem extends React.Component {
+  render() {
+    const { revision, recipe } = this.props;
+    const isCurrent = revision.recipe.revision_id === recipe.revision_id;
```

Comment:
I added some PropTypes to this but it didn't really track down the issue. But it was good to add. I have some open questions about proptypes I'll be asking later.

Buuuuuuuuut I suspect I know the problem. @rehandalal , have you updated your Python requirements lately? If you haven't, you probably don't have the latest reversion, which changed the property that the version instance is stored in for revisions from `revision.object_version` to `revision._object_version`.

I'm willing to bet that the serializer silently fails when it can't find the source for a property, and thus the recipe field from the `RecipeVersionSerializer` is empty.

Which sucks. It shouldn't be silently failing to serialize that. If that's the case it's a bug.